app/controllers/topics_controller.rb

```diff
@@ -58,8 +59,18 @@ def show
     username_filters = opts[:username_filters]
 
     opts[:slow_platform] = true if slow_platform?
+    opts[:print] = true if params[:print].present?
     opts[:username_filters] = username_filters.split(',') if username_filters.is_a?(String)
 
+    if opts[:print]
+      raise Discourse::InvalidAccess unless SiteSetting.allow_print_page
+      begin
+        RateLimiter.new(current_user, "print-topic-per-hour", 10, 1.hour).performed! unless current_user.admin?
```

Comment:
Exactly this! A site setting for rate limit (max prints per hour, default 0) and if max == 0 it disables the feature and removes the button.